### ISSUE TITLE
Not all form validation constraints are being being invoked

### ISSUE THREAD

**Issue by DavidPetrasek** [symfony/symfony#53743]
### Symfony version(s) affected

6.3.8

### Description

Both `UserType` and `AddressType` have constraints set on some of its properties.
The problem: When I submit `MainFormType` only constraints of `AddressType` are being invoked.

Additional info: When the BaseFormType is used in a CollectionType, all constraints are invoked
```
->add('some_collection', CollectionType::class, 
[
       'entry_type' => BaseFormType::class,
       'entry_options' => ['label' => false],
       'allow_add' => true,
]) 
```

### How to reproduce

```
class MainFormType extends AbstractType
{        
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {   
        $builder->add('submit', SubmitType::class);  
    }

    public function getParent(): string
    {
        return BaseFormType::class;
    }
}
```

```
class BaseFormType extends AbstractType
{
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
        $UserType = $builder->create('user', UserType::class,
        [
            'data' => $options['user'],
//         'constraints' => [new Constraints\Valid()],     <--- Tried, not working
//         'constraints' => [new Constraints\Cascade()]    <--- Tried, not working
        ]);
        $UserType->add('address', AddressType::class);
          
        $builder->add($UserType);
    }

    public function configureOptions(OptionsResolver $resolver): void
    {
        $resolver->setDefaults(
        [
            'user' => null
        ]);
    }
}
```

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by DavidPetrasek** [comment]
Maybe later. Now, I do not have the time.

**Comment by xabbuh** [comment]
Let's close here then for now. We can always reopen once we are able to reproduce. Thank you for understanding.

**Comment by DavidPetrasek** [comment]
After trying to make a bug reproducer **I've solved it. It was not a Symfony bug.**

It was giving me this error: `Expected argument of type "string", "null" given at property path "name".`
I just needed to allow null values in the setter of this property in its entity class.

**However,** It really took me a long time to realize that it was related to an entity. So I propose that this error message should also contain the name of the entity like this:
`Expected argument of type "string", "null" given at property path "name" in App\Entity\User.`
